backend/src/routes/profileRoutes.ts

```diff
@@ -0,0 +1,15 @@
+import express, { NextFunction, Request, Response } from "express";
+import ProfileController from "../controllers/profileController";
+
+const router = express.Router();
+
+router.get("/", ProfileController.getProfiles);
+router.get("/:id", ProfileController.getProfile);
+router.post("/", ProfileController.createProfile);
+
+router.use((err: Error, req: Request, res: Response, next: NextFunction) => {
+  console.error(err.stack);
+  res.status(500).json({ message: 'An uncaught error occurred' });
```

Comment:
So, it's better to control error messages from parts when you actually throw it.  Every validation errors would be sent as 500 errors, so your frontend would never guess what's wrong with the request